rust/private/rust.bzl

```diff
@@ -266,6 +266,7 @@ def _rust_library_common(ctx, crate_type):
             edition = get_edition(ctx.attr, toolchain),
             rustc_env = ctx.attr.rustc_env,
             is_test = False,
+            compile_data = depset(ctx.files.compile_data),
```

Comment:
Could I kindly nerd snipe you into creating a starlark unit test that in general would inspect providers returned by a `rust_library`, but you'll only have to add a test that `rust_library` using `compile_data` provides `CrateInfo` with that `compile_data`. I think having more coverage for data in providers is valuable, and It'll be great to have this test started, so it is simpler for other contributors to add another test there when needed.

Reply:
Added a test